tracer/src/Datadog.Trace/Debugger/SpanCodeOrigin/SpanCodeOriginManager.cs

```diff
@@ -0,0 +1,152 @@
+
+            Instance.AddExitSpanTag(span);
+        }
+
+        private void AddExitSpanTag(Span span)
+        {
+            var frames = ArrayPool<FrameInfo>.Shared.Rent(this._settings.CodeOriginMaxUserFrames);
+            try
+            {
+                var framesLength = PopulateUserFrames(frames);
+                if (framesLength == 0)
+                {
+                    Log.Warning("No user frames has founded");
+                    return;
+                }
+
+                span.Tags.SetTag($"{CodeOriginTag}.type", "exit");
+                for (var i = 0; i < framesLength; i++)
+                {
+                    ref var info = ref frames[i];
+
+                    // PopulateUserFrames returns only frames that have method
+                    var method = info.Frame.GetMethod()!;
+                    var type = method.DeclaringType?.FullName ?? method.DeclaringType?.Name;
+                    if (string.IsNullOrEmpty(type))
+                    {
+                        continue;
+                    }
+
+                    span.Tags.SetTag($"{CodeOriginTag}.{FramesPrefix}.{i}.index", info.FrameIndex.ToString());
+                    span.Tags.SetTag($"{CodeOriginTag}.{FramesPrefix}.{i}.method", method.Name);
+                    span.Tags.SetTag($"{CodeOriginTag}.{FramesPrefix}.{i}.type", type);
+
+                    var fileName = info.Frame.GetFileName();
+                    if (!string.IsNullOrEmpty(fileName))
+                    {
+                        // todo: should we normalize?
+                        span.Tags.SetTag($"{CodeOriginTag}.{FramesPrefix}.{i}.file", fileName);
+                    }
+
+                    var line = info.Frame.GetFileLineNumber();
+                    if (line > 0)
+                    {
+                        span.Tags.SetTag($"{CodeOriginTag}.{FramesPrefix}.{i}.line", line.ToString());
+                    }
+
+                    var column = info.Frame.GetFileColumnNumber();
+                    if (column > 0)
+                    {
+                        span.Tags.SetTag($"{CodeOriginTag}.{FramesPrefix}.{i}.column", column.ToString());
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                Log.Error(ex, "Failed to create exit span tag for {Span}", span.SpanId);
+            }
+            finally
+            {
+                // ReSharper disable once ConditionIsAlwaysTrueOrFalseAccordingToNullableAPIContract
+                if (frames != null)
+                {
+                    ArrayPool<FrameInfo>.Shared.Return(frames);
+                }
+            }
+        }
+
+        private int PopulateUserFrames(FrameInfo[] frames)
+        {
+            var stackTrace = new StackTrace(true);
+            var stackFrames = stackTrace.GetFrames();
+
+            // ReSharper disable once ConditionIsAlwaysTrueOrFalseAccordingToNullableAPIContract
+            if (stackFrames == null)
```

Comment:
You can use the `null!` syntax instead of the ReSharper comment.
```suggestion
            if (stackFrames == null!)
```

Reply:
Fixed in 665d7c7e4e719a883d4a4f29ce77930abc8a12df